detection_rules/rule.py

```diff
@@ -220,9 +230,45 @@ class QueryValidator:
     def ast(self) -> Any:
         raise NotImplementedError
 
+    @property
+    def unique_fields(self) -> Any:
+        raise NotImplementedError
+
     def validate(self, data: 'QueryRuleData', meta: RuleMeta) -> None:
         raise NotImplementedError()
 
+    @cached
+    def get_required_fields(self, index: str) -> List[dict]:
+        """retrieves fields needed for the query along with type information from the schema."""
+        current_version = Version(Version(load_current_package_version()) + (0,))
```

Comment:
@brokensound77 this will break users with stack `7.17` when trying to run commands like `python3 -m detection_rules kibana upload-rule`

Reply:
@brokensound77 This is because 7.17 cannot be backported to across major versions (8 to 7) and secondly because the `stack-schema-map.yml` does not have a `7.17.0` entry. Resolving as out of scope.